auth/plain.go

```diff
@@ -11,7 +22,15 @@ type Plain struct {
 
 func (p *Plain) Check(c server.ClientAuth) bool {
 	opts := c.GetOpts()
-	if p.Username != opts.Username || p.Password != opts.Password {
+	if p.Username != opts.Username {
+		return false
+	}
+	// Check to see if the password is a bcrypt hash
+	if isBcrypt(p.Password) {
+		if err := bcrypt.CompareHashAndPassword([]byte(p.Password), []byte(opts.Password)); err != nil {
+			return false
```

Comment:
The issue is around operational pain for the people setting this up.  Exposing systemic problems instead of throwing away that information matters for making products easy to deploy.  Correctness of result is important but "in the end" is a poor approach to deployability and debugability.